src/descriptor/key_map.rs

```diff
@@ -0,0 +1,276 @@
+use bitcoin::psbt::{GetKey, GetKeyError, KeyRequest};
+use bitcoin::secp256k1::{Secp256k1, Signing};
+use bitcoin::PrivateKey;
+
+use crate::descriptor::{DescriptorSecretKey, KeyMap};
+use crate::BTreeMap;
+
+/// A wrapper around KeyMap that implements GetKey for PSBT signing.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct KeyMapWrapper {
+    map: KeyMap,
+}
+
+impl From<KeyMap> for KeyMapWrapper {
+    fn from(map: KeyMap) -> Self { KeyMapWrapper { map } }
+}
+
+impl GetKey for KeyMapWrapper {
+    type Error = GetKeyError;
+
+    fn get_key<C: Signing>(
+        &self,
+        key_request: KeyRequest,
+        secp: &Secp256k1<C>,
+    ) -> Result<Option<bitcoin::PrivateKey>, Self::Error> {
+        Ok(self
+            .map
+            .iter()
+            .find_map(|(_desc_pk, desc_sk)| -> Option<PrivateKey> {
+                match desc_sk.get_key(key_request.clone(), secp) {
+                    Ok(Some(pk)) => Some(pk),
+                    Ok(None) | Err(_) => None,
```

Comment:
In c7d41712ccc581a8fb72eb8e4e1861e4f56b00d1:

Here (and below) you swallow the error from `DescriptorSecretKey::get_key` and turn it into a "key not found" by returning `None`.

This is an issue with the code on master. I will open a fix PR and then I'd like you to also backport that.

Reply:
Lemme PR some unit tests and I'll just ask you to backport those :P.

Reply:
Thanks, I've cherry-picked the commits and updated the required types for `KeyMapWrapper`.